pkg/processor/k8sprocessor/kube/client.go

```diff
@@ -380,6 +377,14 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	return tags
 }
 
+func (c *WatchClient) updatePodServices(pod *Pod) {
+	if c.Rules.ServiceName {
+		if services := c.op.GetServices(pod.Name); len(services) > 0 {
```

Comment:
if there is no services, we should set empty string or remove the attribute

Reply:
Out of curiosity. can we shorten it to just

```
pod.Attributes[c.Rules.Tags.ServiceName] = strings.Join(services, c.delimiter)
```

?

Reply:
According to the implementation, yes: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/strings/strings.go;l=430
I did not change it, because I was unsure if such way of coding is idiomatic in Go. But I also like it better.

Reply:
I would go with it